milestones/serializers.py

```diff
@@ -0,0 +1,67 @@
+"""
+Data layer serialization operations.  Converts querysets to simple
+python containers (mainly arrays and dicts).
+"""
+from . import models
+
+
+def serialize_milestone(milestone):
+    """
+    Milestone object-to-dict serialization
+    """
+    return {
+        'id': milestone.id,
+        'name': milestone.name,
+        'namespace': milestone.namespace,
+        'description': milestone.description
+    }
+
+
+def serialize_milestone_with_course(course_milestone):
+    """
+    CourseMilestone serialization (composite object)
+    """
+    return {
+        'id': course_milestone.milestone.id,
+        'name': course_milestone.milestone.name,
+        'namespace': course_milestone.milestone.namespace,
+        'description': course_milestone.milestone.description,
+        'course_id': course_milestone.course_id
+    }
+
+
+def serialize_milestone_with_course_content(course_content_milestone):
+    """
+    CourseContentMilestone serialization (composite object)
+    """
+    return {
+        'id': course_content_milestone.milestone.id,
+        'name': course_content_milestone.milestone.name,
+        'namespace': course_content_milestone.milestone.namespace,
+        'description': course_content_milestone.milestone.description,
+        'course_id': course_content_milestone.course_id,
+        'content_id': course_content_milestone.content_id
+    }
+
+
+def serialize_milestones(milestones):
```

Comment:
I spent a couple hours trying to transition over to using DRF serializers/validators.  Maybe I'm just thick but I wasn't able to get it to work, so I rolled back to the custom stuff.  I think this refactor should go in as a fast follower -- ie, part of our Q3 work (@marcotuts, FYI)